Added option to define text separators to use

ufobm opened the pull request.

Commit:
Added an option to specify the text separators to use

Commit:
Added option to define text separators to use

eyoeldefare commented:
Thanks for the contribution, I will review this very soon.

eyoeldefare committed:
fixes issue #31 - missed null check

Commit:
add icon feature about #32 (#33)

eyoeldefare commented:
Hey there, tests haves failed for entering tags with comma, and entering tags with space-bar out of the box. What exactly is this trying to achieve and please test all functionalities locally first before merge request. 
Thanks for contributing

eyoeldefare committed:
optimise for mem

eyoeldefare committed:
release v1.4.0

ufobm committed:
Merge branch 'add_text_separators'

# Conflicts:
#	lib/src/main.dart

ufobm committed:
Merge branch 'master' of https://github.com/ufobm/textfield_tags

# Conflicts:
#	lib/src/main.dart

ufobm committed:
Restored default text separators

ufobm commented:
Are you talking about manual tests? I don't think automatic tests are implemented. Of course, comma and space-bar remains the defaults text separators! I added an option to define different text separators, like so:

textSeparators: [".", "!"],

I removed this line from the example, so now it should work as expected out of the box.

eyoeldefare commented:
Yes, I am manually testing the functionalities all around. 
To cut down on time, before I examine the changes, I simply check if the app works out of the box with new changes. So i appreciate a little description on what new pull requests do.
Regardless, I will test this again over this weekend. 
Thanks

eyoeldefare commented on lib/src/main.dart:
```diff
@@ -231,18 +240,22 @@ class _TextFieldTagsState extends State<TextFieldTags> {
               if (!_tagsStringContents!.contains(lastLastTag)) {
-                if (widget.validator == null ||
-                    widget.validator!(lastLastTag) == null) {
+                final validatorResult = widget.validator!(lastLastTag);
```
Reverse this because if user doesn't implement `validator`, it gives of `null` where you can't `null!`.

ufobm replied:
Done! Thank you!

eyoeldefare reviewed (commented):
Overall, looks good to me.

ufobm committed:
If user doesn't implement validator, now it doesn't give null where you can't null!

ufobm commented:
Thank you! Will this merge be published on pub dev?

eyoeldefare commented:
hey, yeah I will publish it this weekend. 
Eyoel
